CedarJava/src/main/java/com/cedarpolicy/SliceJsonSerializer.java

```diff
@@ -58,15 +58,34 @@ private static class JsonEntity {
         public final Set<JsonEUID> parents;
 
         JsonEntity(Entity e) {
-            String[] uid_parts = e.uid.split("::");
-            String uid_id = uid_parts[uid_parts.length-1].substring(1,uid_parts[uid_parts.length-1].length()-1);
-            String uid_type = e.uid.substring(0, e.uid.length()-uid_id.length()-4);
+            if(!e.getEuid().isEmpty()) {
+                this.uid = e.getEuid().get();
+            } else {
+                String[] uid_parts = e.uid.split("::");
+                //Types cannot be empty but ids can. If the string ends with `:::` the id is `":"`. If the string ends with `::` the id is `""` and everything before the last `::` is the type.
```

Comment:
What happened to the quotes? Shouldn't these cases look like `EntityType::":"` and `EntityType::""`?

Reply:
Correct behavior comes from line 61 above. At this point in the code, the extra quotes around the id don't exist, so there isn't a way to handle every case correctly if we reach this else block.

We should ensure we always hit line 62, but I think we should merge this now since it's more correct than the status quo and then fix that later.

Reply:
My comments aren't merge blocking, was just curious what this was for 🙂